### DISCUSSION TITLE
Row selection doesn't work on grid unless the underlying model has an id.

### DISCUSSION THREAD

**Discussion by tellingson** [IgniteUI/igniteui-angular#14177] [Q&A] [answered]
## Description  
If the underlying model for a grid row has an ID (appears to not be case sensitive) row selection doesn't work.

 * igniteui-angular version:  latest
 * browser: Firefox

## Steps to reproduce  

1. Go to https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/row-selection
2. Open the stackblitz example 
3. Change the ID in the financialData.ts to another property. (Foo)
4. Update the grid-selection.component.html primary key to that property [primaryKey]="'Foo'"
5. Test the component.

## Result  
What is the actual result after following the steps to reproduce?
Notice that any selection will select all options. If you listen to the events, all rows will get selected.
If you go back to the financialData.ts and add the ID (but keep your other change also adding foo
 dataObj.Foo = i;
 dataObj.ID = i;
Notice that row selection now works. I have no idea if it is respecting the primary key or if it is just using ID.

## Expected result  
What is the expected result after following the steps to reproduce?
I expect to not be required to have ID be an attribute of any models that I want to use within the grid. I also expect to be able to set the primary key on the row.

## Attachments  
Attach a sample if available, and screenshots, if applicable.

**Comment by georgianastasov** [comment, marked as answer]
Hello @tellingson,

After carefully following the steps provided, I attempted to replicate the behavior described, but regrettably, I could not reproduce the anomaly on my end. On my side everything works as expected and I am able to select each row in the igx-grid component.

The described scenario could be observed here:
![github1](https://github.com/IgniteUI/igniteui-angular/assets/48180072/b630acc4-400a-4cb9-9c33-a09309192779)


[Here](https://stackblitz.com/edit/e8z9yd?file=src%2Fapp%2Fgrid%2Fgrid-sample-selection%2Fgrid-selection.component.html) you will find my sample for your reference. 

Upon further analysis, I made an assumption that the discrepancy might stem from a potential oversight in the alteration of the id property to foo, maybe you didn't changed it everywhere within the financialData.ts file, as per your instructions.

Considering this assumption, it's imperative for me to delve deeper into the matter to pinpoint the exact cause of the observed behavior. To facilitate a more thorough investigation, I kindly request that you provide me with an isolated sample that accurately reproduces the behavior described you're experiencing.
